samples/fs/littlefs : qemu_x86 crashes with the compiled binary

**Describe the bug**
The compiled binary for the samples/fs/littlefs for board qemu_x86 crashes.
The execution goes through perfectly till the end, and all the API's are executed as expected, but right at the end of the program the system crashes.

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p=auto -b qemu_x86 samples/subsys/fs/littlefs/
2. west build -t run
4. See error

**Expected behavior**
A clean output of all the littlefs API's, without any crash.

**Impact**
Needed to make it work with qemu_cortex_r5, but instead I'm stuck at fixing this. 

**Logs and console output**
```
ali@ubuntu:~/zephyrproject/zephyr$ west build -p=auto -b qemu_x86 samples/subsys/fs/littlefs/

-- west build: making build dir /home/ali/zephyrproject/zephyr/build pristine

-- west build: generating a build system

Loading Zephyr default modules (Zephyr base).

-- Application: /home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs

-- CMake version: 3.22.1

-- Found Python3: /usr/bin/python3.10 (found suitable exact version "3.10.6") found components: Interpreter 

-- Cache files will be written to: /home/ali/.cache/zephyr

-- Zephyr version: 3.3.99 (/home/ali/zephyrproject/zephyr)

-- Found west (found suitable version "1.0.0", minimum required is "0.7.1")

-- Board: qemu_x86

-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK

-- Found host-tools: zephyr 0.16.0 (/home/ali/zephyr-sdk-0.16.0)

-- Found toolchain: zephyr 0.16.0 (/home/ali/zephyr-sdk-0.16.0)

-- Found Dtc: /home/ali/zephyr-sdk-0.16.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 

-- Found BOARD.dts: /home/ali/zephyrproject/zephyr/boards/x86/qemu_x86/qemu_x86.dts

-- Generated zephyr.dts: /home/ali/zephyrproject/zephyr/build/zephyr/zephyr.dts

-- Generated devicetree_generated.h: /home/ali/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h

-- Including generated dts.cmake file: /home/ali/zephyrproject/zephyr/build/zephyr/dts.cmake

Parsing /home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs/Kconfig

Loaded configuration '/home/ali/zephyrproject/zephyr/boards/x86/qemu_x86/qemu_x86_defconfig'

Merged configuration '/home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs/prj.conf'

Configuration saved to '/home/ali/zephyrproject/zephyr/build/zephyr/.config'

Kconfig header saved to '/home/ali/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'

-- Found GnuLd: /home/ali/zephyr-sdk-0.16.0/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/12.2.0/../../../../x86_64-zephyr-elf/bin/ld.bfd (found version "2.38") 

-- The C compiler identification is GNU 12.2.0

-- The CXX compiler identification is GNU 12.2.0

-- The ASM compiler identification is GNU

-- Found assembler: /home/ali/zephyr-sdk-0.16.0/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc

CMake Warning at /home/ali/zephyrproject/zephyr/subsys/random/CMakeLists.txt:11 (message):

  



      Warning: CONFIG_TIMER_RANDOM_GENERATOR is not a truly random generator.

      This capability is not secure and it is provided for testing purposes only.

      Use it carefully.





-- Configuring done

-- Generating done

-- Build files have been written to: /home/ali/zephyrproject/zephyr/build

-- west build: building application

[1/150] Preparing syscall dependency handling



[3/150] Generating include/generated/version.h

-- Zephyr version: 3.3.99 (/home/ali/zephyrproject/zephyr), build: zephyr-v3.3.0-3420-g7a946450e51e

[136/150] Linking C executable zephyr/zephyr_pre0.elf



[140/150] Linking C executable zephyr/zephyr_pre1.elf



[150/150] Linking C executable zephyr/zephyr.elf

Memory region         Used Size  Region Size  %age Used

             RAM:     1171488 B         3 MB     37.24%

        IDT_LIST:          0 GB         2 KB      0.00%

ali@ubuntu:~/zephyrproject/zephyr$ 

ali@ubuntu:~/zephyrproject/zephyr$ 

ali@ubuntu:~/zephyrproject/zephyr$ 

ali@ubuntu:~/zephyrproject/zephyr$ west build -t run

-- west build: running target run

[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: qemu32,+nx,+pae

SeaBIOS (version zephyr-v1.0.0-0-g31d4e0e-dirty-20200714_234759-fv-az50-zephyr)

Booting from ROM..

*** Booting Zephyr OS build zephyr-v3.3.0-3420-g7a946450e51e ***

Sample program to r/w files on littlefs

Area 1 at 0x1000 on sim_flash for 65536 bytes

I: LittleFS version 2.5, disk version 2.0

I: FS at sim_flash:0x1000 is 64 0x400-byte blocks with 512 cycle

I: sizes: rd 16 ; pr 16 ; ca 64 ; la 32

E: WEST_TOPDIR/modules/fs/littlefs/lfs.c:1234: Corrupted dir pair at {0x0, 0x1}

W: can't mount (LFS -84); formatting

I: /lfs mounted

/lfs mount: 0

/lfs: bsize = 16 ; frsize = 1024 ; blocks = 64 ; bfree = 62



Listing dir /lfs ...

/lfs/boot_count read count:0 (bytes: 0)

/lfs/boot_count write new boot count 1: [wr:1]

I: Test file: /lfs/pattern.bin not found, create one!

------ FILE: /lfs/pattern.bin ------

01 55 55 55 55 55 55 55 02 55 55 55 55 55 55 55

03 55 55 55 55 55 55 55 04 55 55 55 55 55 55 55

05 55 55 55 55 55 55 55 06 55 55 55 55 55 55 55

07 55 55 55 55 55 55 55 08 55 55 55 55 55 55 55

09 55 55 55 55 55 55 55 0a 55 55 55 55 55 55 55

0b 55 55 55 55 55 55 55 0c 55 55 55 55 55 55 55

0d 55 55 55 55 55 55 55 0e 55 55 55 55 55 55 55

0f 55 55 55 55 55 55 55 10 55 55 55 55 55 55 55

11 55 55 55 55 55 55 55 12 55 55 55 55 55 55 55

13 55 55 55 55 55 55 55 14 55 55 55 55 55 55 55

15 55 55 55 55 55 55 55 16 55 55 55 55 55 55 55

17 55 55 55 55 55 55 55 18 55 55 55 55 55 55 55

19 55 55 55 55 55 55 55 1a 55 55 55 55 55 55 55

1b 55 55 55 55 55 55 55 1c 55 55 55 55 55 55 55

1d 55 55 55 55 55 55 55 1e 55 55 55 55 55 55 55

1f 55 55 55 55 55 55 55 20 55 55 55 55 55 55 55

21 55 55 55 55 55 55 55 22 55 55 55 55 55 55 55

23 55 55 55 55 55 55 55 24 55 55 55 55 55 55 55

25 55 55 55 55 55 55 55 26 55 55 55 55 55 55 55

27 55 55 55 55 55 55 55 28 55 55 55 55 55 55 55

29 55 55 55 55 55 55 55 2a 55 55 55 55 55 55 55

2b 55 55 55 55 55 55 55 2c 55 55 55 55 55 55 55

2d 55 55 55 55 55 55 55 2e 55 55 55 55 55 55 55

2f 55 55 55 55 55 55 55 30 55 55 55 55 55 55 55

31 55 55 55 55 55 55 55 32 55 55 55 55 55 55 55

33 55 55 55 55 55 55 55 34 55 55 55 55 55 55 55

35 55 55 55 55 55 55 55 36 55 55 55 55 55 55 55

37 55 55 55 55 55 55 55 38 55 55 55 55 55 55 55

39 55 55 55 55 55 55 55 3a 55 55 55 55 55 55 55

3b 55 55 55 55 55 55 55 3c 55 55 55 55 55 55 55

3d 55 55 55 55 55 55 55 3e 55 55 55 55 55 55 55

3f 55 55 55 55 55 55 55 40 55 55 55 55 55 55 55



41 55 55 55 55 55 55 55 42 55 55 55 55 55 55 55

43 55 55 55 55 55 55 55 44 55 55 55 55 55 55 55

45 55 aa 

I: /lfs unmounted

/lfs unmount: 0

E: Page fault at address 0x4 (error code 0x10)

E: Linear address not present in page tables

E: Access violation: supervisor thread not allowed to execute

E: PTE: not present

E: EAX: 0x00000000, EBX: 0x00000000, ECX: 0x00215ce0, EDX: 0x00215ce0

E: ESI: 0x00218994, EDI: 0x00000004, EBP: 0x00000000, ESP: 0x002180b4

E: EFLAGS: 0x00000202 CS: 0x0008 CR3: 0x0021e000

E: call trace:

E: EIP: 0x00000004

E: NULL base ptr

E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0

E: Current thread: 0x215c80 (unknown)

E: Halting system

FAILED: zephyr/CMakeFiles/run_qemu /home/ali/zephyrproject/zephyr/build/zephyr/CMakeFiles/run_qemu 

cd /home/ali/zephyrproject/zephyr/build && /home/ali/zephyr-sdk-0.16.0/sysroots/x86_64-pokysdk-linux/usr/bin/qemu-system-i386 -m 4 -cpu qemu32,+nx,+pae -machine q35 -device isa-debug-exit,iobase=0xf4,iosize=0x04 -no-reboot -nographic -no-acpi -net none -pidfile qemu.pid -chardev stdio,id=con,mux=on -serial chardev:con -mon chardev=con,mode=readline -icount shift=5,align=off,sleep=off -rtc clock=vm -kernel /home/ali/zephyrproject/zephyr/build/zephyr/zephyr.elf

ninja: build stopped: subcommand failed.

FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/ali/zephyrproject/zephyr/build --target run

ali@ubuntu:~/zephyrproject/zephyr$ 
```

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04.1
 - Toolchain: default
 - the latest one as of today

**Additional context**
Any help regarding how to make it the example work with qemu_cortex_r5, since its build failing with a different error which has got to do the flash emulation and the partitioning, AFAIK.

> ali@ubuntu:~/zephyrproject/zephyr$ west build -p=auto -b qemu_cortex_r5 samples/subsys/fs/littlefs/
> 
> -- west build: generating a build system
> 
> Loading Zephyr default modules (Zephyr base).
> 
> -- Application: /home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs
> 
> -- CMake version: 3.22.1
> 
> -- Found Python3: /usr/bin/python3.10 (found suitable exact version "3.10.6") found components: Interpreter 
> 
> -- Cache files will be written to: /home/ali/.cache/zephyr
> 
> -- Zephyr version: 3.3.99 (/home/ali/zephyrproject/zephyr)
> 
> -- Found west (found suitable version "1.0.0", minimum required is "0.7.1")
> 
> -- Board: qemu_cortex_r5
> 
> -- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
> 
> -- Found host-tools: zephyr 0.16.0 (/home/ali/zephyr-sdk-0.16.0)
> 
> -- Found toolchain: zephyr 0.16.0 (/home/ali/zephyr-sdk-0.16.0)
> 
> -- Found Dtc: /home/ali/zephyr-sdk-0.16.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
> 
> -- Found BOARD.dts: /home/ali/zephyrproject/zephyr/boards/arm/qemu_cortex_r5/qemu_cortex_r5.dts
> 
> -- Generated zephyr.dts: /home/ali/zephyrproject/zephyr/build/zephyr/zephyr.dts
> 
> -- Generated devicetree_generated.h: /home/ali/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h
> 
> -- Including generated dts.cmake file: /home/ali/zephyrproject/zephyr/build/zephyr/dts.cmake
> 
> 
> 
> warning: FLASH_PAGE_LAYOUT (defined at drivers/flash/Kconfig:80) was assigned the value 'y' but got
> 
> the value 'n'. Check these unsatisfied dependencies: FLASH_HAS_PAGE_LAYOUT (=n). See
> 
> http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_FLASH_PAGE_LAYOUT and/or look up
> 
> FLASH_PAGE_LAYOUT in the menuconfig/guiconfig interface. The Application Development Primer, Setting
> 
> Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
> 
> too.
> 
> 
> 
> 
> 
> warning: FILE_SYSTEM_LITTLEFS (defined at subsys/fs/Kconfig.littlefs:6) was assigned the value 'y'
> 
> but got the value 'n'. Check these unsatisfied dependencies: FLASH_MAP (=n), FLASH_PAGE_LAYOUT (=n).
> 
> See http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_FILE_SYSTEM_LITTLEFS and/or look up
> 
> FILE_SYSTEM_LITTLEFS in the menuconfig/guiconfig interface. The Application Development Primer,
> 
> Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
> 
> helpful too.
> 
> 
> 
> 
> 
> warning: FLASH_MAP (defined at subsys/storage/flash_map/Kconfig:10) was assigned the value 'y' but
> 
> got the value 'n'. Check these unsatisfied dependencies: FLASH_HAS_DRIVER_ENABLED (=n). See
> 
> http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_FLASH_MAP and/or look up FLASH_MAP in the
> 
> menuconfig/guiconfig interface. The Application Development Primer, Setting Configuration Values,
> 
> and Kconfig - Tips and Best Practices sections of the manual might be helpful too.
> 
> 
> 
> Parsing /home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs/Kconfig
> 
> Loaded configuration '/home/ali/zephyrproject/zephyr/boards/arm/qemu_cortex_r5/qemu_cortex_r5_defconfig'
> 
> Merged configuration '/home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs/prj.conf'
> 
> Configuration saved to '/home/ali/zephyrproject/zephyr/build/zephyr/.config'
> 
> Kconfig header saved to '/home/ali/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
> 
> -- Found GnuLd: /home/ali/zephyr-sdk-0.16.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd (found version "2.38") 
> 
> -- The C compiler identification is GNU 12.2.0
> 
> -- The CXX compiler identification is GNU 12.2.0
> 
> -- The ASM compiler identification is GNU
> 
> -- Found assembler: /home/ali/zephyr-sdk-0.16.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
> 
> -- Configuring done
> 
> -- Generating done
> 
> -- Build files have been written to: /home/ali/zephyrproject/zephyr/build
> 
> -- west build: building application
> 
> [1/142] Preparing syscall dependency handling
> 
> 
> 
> [3/142] Generating include/generated/version.h
> 
> -- Zephyr version: 3.3.99 (/home/ali/zephyrproject/zephyr), build: zephyr-v3.3.0-3420-g7a946450e51e
> 
> [15/142] Building C object CMakeFiles/app.dir/src/main.c.obj
> 
> FAILED: CMakeFiles/app.dir/src/main.c.obj 
> 
> ccache /home/ali/zephyr-sdk-0.16.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/ali/zephyrproject/zephyr/include -I/home/ali/zephyrproject/zephyr/build/zephyr/include/generated -I/home/ali/zephyrproject/zephyr/soc/arm/xilinx_zynqmp -I/home/ali/zephyrproject/modules/hal/cmsis/CMSIS/Core_R/Include -isystem /home/ali/zephyrproject/zephyr/lib/libc/minimal/include -isystem /home/ali/zephyr-sdk-0.16.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/include -isystem /home/ali/zephyr-sdk-0.16.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/include-fixed -fno-strict-aliasing -Og -imacros /home/ali/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-r5+nofp -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/home/ali/zephyr-sdk-0.16.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/ali/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/ali/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/ali/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj -c /home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs/src/main.c
> 
> In file included from /home/ali/zephyrproject/zephyr/samples/subsys/fs/littlefs/src/main.c:15:
> 
> /home/ali/zephyrproject/zephyr/include/zephyr/fs/littlefs.h:14:10: fatal error: lfs.h: No such file or directory
> 
>    14 | #include <lfs.h>
> 
>       |          ^~~~~~~
> 
> compilation terminated.
> 
> [24/142] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
> 
> ninja: build stopped: subcommand failed.
> 
> FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/ali/zephyrproject/zephyr/build
> 
> ali@ubuntu:~/zephyrproject/zephyr$

--- a/samples/subsys/fs/littlefs/prj.conf
+++ b/samples/subsys/fs/littlefs/prj.conf
@@ -8,7 +8,7 @@
 #CONFIG_APP_WIPE_STORAGE=y
 
 # fs_dirent structures are big.
-CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_MAIN_STACK_SIZE=4096
 
 # Let __ASSERT do its job
 CONFIG_DEBUG=y

